app/packages/spotlight/src/createScrollReader.ts

```diff
@@ -73,6 +74,14 @@ export default function createScrollReader(
       destroyed = true;
       element.removeEventListener("scroll", scroll);
       element.removeEventListener("scrollend", scrollEnd);
+
+      if (timeout) {
+        clearTimeout(timeout);
+      }
+
+      if (animationFrameId) {
+        cancelAnimationFrame(animationFrameId);
+      }
```

Comment:
It seems like there is redundancy here. I would be surprised if the closure here is causing a memory leak. But it doesn't hurt 👍